File: Ransome-note-problem/ransome-note-solution.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    if ( ransomNote.length > magazine.length ) return false;
    const oldMagazineLength = magazine.length;
    ransomNote.split('').forEach(item => {
        magazine = magazine.replace(item, '');
    });
     if(oldMagazineLength == magazine.length + ransomNote.length) {
        return true;
    } else {
        return false;
    }
};

File: Ransome-note-problem/ransome-note-solution.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        dic = {}
        for r in ransomNote:
            dic[r] = ransomNote.count(r)
        for key in dic.keys():
            if magazine.count(key) < dic[key]:
                return False
        return True
        
        